executor/common_linux.h

```diff
@@ -4285,3 +4318,194 @@ static void setup_usb()
 // The macro is used in generated C code.
 #define CAST(f) ({void* p = (void*)f; p; })
 #endif
+
+#if SYZ_EXECUTOR || __NR_syz_fuse_handle_req
+#include <fcntl.h>
+#include <linux/fuse.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+// Struct shared between syz_fuse_handle_req() and the fuzzer. Used to provide
+// a fuzzed response for each request type.
+struct syz_fuse_req_out {
+	struct fuse_out_header* init;
+	struct fuse_out_header* lseek;
+	struct fuse_out_header* bmap;
+	struct fuse_out_header* poll;
+	struct fuse_out_header* getxattr;
+	struct fuse_out_header* lk;
+	struct fuse_out_header* statfs;
+	struct fuse_out_header* write;
+	struct fuse_out_header* read;
+	struct fuse_out_header* open;
+	struct fuse_out_header* attr;
+	struct fuse_out_header* entry;
+	struct fuse_out_header* dirent;
+	struct fuse_out_header* direntplus;
+	struct fuse_out_header* create_open;
+	struct fuse_out_header* ioctl;
+};
+
+// Link the reponse to the request and send it to /dev/fuse.
+static int
+fuse_send_response(int fd,
+		   const struct fuse_in_header* in_hdr,
+		   struct fuse_out_header* out_hdr)
+{
+	int ret;
+
+	if (!out_hdr) {
+		debug("fuse_send_response(): received a NULL out_hdr\n");
+		return -1;
+	}
+
+	out_hdr->unique = in_hdr->unique;
+	ret = write(fd, out_hdr, out_hdr->len);
+	if (ret == -1) {
+		debug("fuse_send_response > write() failed: %d\n", errno);
+		return -1;
+	}
+
+	return 0;
+}
+
+// This function reads a request from /dev/fuse and tries to pick the correct
+// response from the input struct syz_fuse_req_out (a3). Responses are still
+// generated by the fuzzer.
+static volatile long syz_fuse_handle_req(volatile long a0, // /dev/fuse fd.
+					 volatile long a1, // Read buffer.
+					 volatile long a2, // Buffer len.
+					 volatile long a3) // syz_fuse_req_out.
+{
+	const struct fuse_in_header* in_hdr;
+	struct syz_fuse_req_out* req_out;
+	struct fuse_out_header* out_hdr;
+	char* buf;
+	int buf_len;
+	int ret;
+	int fd;
+
+	fd = (int)a0;
+	buf = (char*)a1;
+	buf_len = (int)a2;
+	req_out = (struct syz_fuse_req_out*)a3;
+
+	if (!req_out) {
+		debug("syz_fuse_handle_req(): Received a NULL syz_fuse_req_out\n");
+		return -1;
+	}
+	if (buf_len < FUSE_MIN_READ_BUFFER) {
+		debug("FUSE requires the read buffer to be at least %u\n", FUSE_MIN_READ_BUFFER);
+		return -1;
+	}
+
+	ret = read(fd, buf, buf_len);
+	if (ret == -1) {
+		debug("syz_fuse_handle_req > read() failed: %d\n", errno);
+		return -1;
+	}
+	// Safe to do beacuse ret > 0 (!= -1) and < FUSE_MIN_READ_BUFFER
```

Comment:
We don't  use 80-colon typewriters to work with this code ;)
Don't leaving just few characters on the next line will make it tidier:
```
// Safe to do beacuse ret > 0 (!= -1) and < FUSE_MIN_READ_BUFFER (= 8192).
```
(few lines below you violate 80 chars anyway)

Reply:
I think the comment was initially longer, and then I removed something. Anyway I have updated it.